### PR TITLE
test: new sdk cypress tests

### LINKED ISSUE #1509: test: new sdk cypress tests

### PR DESCRIPTION
**Pull request by preetamrevankar**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [x] CI/CD

## Description

This PR adds comprehensive Cypress E2E tests for the SDK, organized into two main test directories:
https://github.com/juspay/hyperswitch-web/issues/1509

### 01-sdk-core (SDK Core Tests)
Tests for core SDK functionality:
- **branding-visibility.cy.ts** - Tests for SDK branding and logo visibility
- **keyboard-accessibility.cy.ts** - Keyboard navigation and accessibility tests
- **layout-accordion.cy.ts** - Accordion layout rendering and interaction tests
- **layout-grid-tabs.cy.ts** - Grid tabs layout tests
- **layout-spaced-accordion.cy.ts** - Spaced accordion layout tests
- **layout-tabs.cy.ts** - Tabs layout rendering tests
- **locale-i18n.cy.ts** - Internationalization and locale tests
- **sdk-error-handling-test.cy.ts** - SDK error handling scenarios
- **submit-button-states.cy.ts** - Submit button state management tests
- **terms-display.cy.ts** - Terms and conditions display tests
- **themes.cy.ts** - Theme customization tests

### 02-cards (Card Payment Tests)
Tests for card payment flows:
- **billing-fields-visibility.cy.ts** - Billing fields visibility and validation
- **card-formatting-paste.cy.ts** - Card number formatting and paste handling
- **card-payment-edge-cases.cy.ts** - Edge cases in card payment flows
- **cobadge-card-flow-cybersource.cy.ts** - Co-badge card flow with Cybersource
- **custom-card-terms.cy.ts** - Custom card terms display tests
- **default-values-prefill.cy.ts** - Default values and prefill functionality
- **dynamic-fields-comprehensive.cy.ts** - Comprehensive dynamic fields tests
- **mandate-card-flow-cybersource.cy.ts** - Mandate card flow with Cybersource

### Other Changes
- Updated cypress support files (`commands.ts`, `utils.ts`, `types.ts`, `e2e.ts`)
- Added new card test data in `cards.ts`
- Updated `cypress.config.js` with new configurations

## How did you test it?

- Ran all Cypress tests locally to verify they pass
- Tested individual test suites for each category

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [x] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/e2e/01-sdk-core/keyboard-accessibility.cy.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  let getIframeBody: () => Cypress.Chainable<JQuery<HTMLBodyElement>>;
+  const iframeSelector =
+    "#orca-payment-element-iframeRef-orca-elements-payment-element-payment-element";
+
+  beforeEach(() => {
+    getIframeBody = () => cy.iframe(iframeSelector);
+    changeObjectKeyValue(
+      createPaymentBody,
+      "profile_id",
+      "pro_5fVcCxU8MFTYozgtf0P8",
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/01-sdk-core/keyboard-accessibility.cy.ts]: Don't hardcode profile ids, store it in `connectorProfileIdMapping`

same for these files too

Image

File: cypress-tests/cypress/e2e/02-cards/mandate-card-flow-cybersource.cy.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      createPaymentBody,
+      "authentication_type",
+      "no_three_ds"
+    );
+    changeObjectKeyValue(createPaymentBody, "customer_id", customerId);
+    changeObjectKeyValue(
+      createPaymentBody,
+      "setup_future_usage",
+      "off_session"
+    );
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/02-cards/mandate-card-flow-cybersource.cy.ts:53]: Use beforeEach to set up all required state (customer, future_usage) before the test that verifies the mandate flow.

File: cypress-tests/cypress/e2e/01-sdk-core/themes.cy.ts [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+        line3: "Harrison Street",
+        city: "San Fransico",
+        state: "California",
+        zip: "94122",
+        country: "US",
+        first_name: "joseph",
+        last_name: "Doe",
+      },
+      phone: { number: "8056594427", country_code: "+91" },
+    });
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/01-sdk-core/themes.cy.ts]: I noticed that the same billing address is hardcoded across multiple files. Could you please create a constant named defaultBillingAddress in a utils.ts file and reuse it wherever needed?

File: cypress-tests/cypress/e2e/01-sdk-core/keyboard-accessibility.cy.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+      getIframeBody()
+        .find(`[data-testid=${testIds.expiryInputTestId}]`)
+        .type(card_exp_month + card_exp_year);
+
+      getIframeBody()
+        .find(`[data-testid=${testIds.cardCVVInputTestId}]`)
+        .type(cvc);
+
+      cy.get("#submit").click();
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/01-sdk-core/keyboard-accessibility.cy.ts]:
```suggestion
      cy.get("#submit").should("be.visible").click();
```
Can we use this wherever you're currently using`cy.get("#submit").click();`?

File: .github/workflows/run-automation-test.yml [hunk 5]
```diff
@@ -7,38 +7,39 @@ on:
...
       - name: Checkout repository
         uses: actions/checkout@v3
-        # Checks out the repository so that the workflow can access the code
 
       - name: Set up Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: "20" # Specifies the version of Node.js to use
-          cache: "npm" # Caches npm dependencies to speed up subsequent installs
+          node-version: "20"
```
Comment (aritro2002) [inline .github/workflows/run-automation-test.yml]: please don't remove the comments of this file, it will make it easier to understand in future.

File: cypress-tests/cypress/e2e/01-sdk-core/locale-i18n.cy.ts [hunk 6]
```diff
@@ -0,0 +1,538 @@
...
+        getIframeBody()
+          .contains(billingDetailsText, { timeout: 10000 })
+          .should("be.visible");
+      });
+    });
+  });
+
+  /* ---------------------------------------------------------------- */
+  /*  6. Fallback behaviour for unsupported locale                     */
+  /* ---------------------------------------------------------------- */
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/01-sdk-core/locale-i18n.cy.ts]: remove such type of comments

File: cypress-tests/cypress/e2e/01-sdk-core/terms-display.cy.ts [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    );
+    changeObjectKeyValue(
+      createPaymentBody,
+      "setup_future_usage",
+      "off_session"
+    );
+  });
+
+  afterEach(() => {
+    delete (createPaymentBody as Record<string, unknown>)["setup_future_usage"];
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/01-sdk-core/terms-display.cy.ts]: use removeObjectKey

File: cypress-tests/cypress/e2e/01-sdk-core/terms-display.cy.ts [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    it("should hide card terms text when card is selected", () => {
+      getIframeBody()
+        .find(".TermsTextLabel")
+        .should("not.exist");
+    });
+  });
+
+  describe("terms.card: not applicable for non-mandate flow", () => {
+    beforeEach(() => {
+      delete (createPaymentBody as Record<string, unknown>)[
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/01-sdk-core/terms-display.cy.ts]: use removeObjectKey

File: cypress-tests/cypress/e2e/02-cards/custom-card-terms.cy.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    );
+    changeObjectKeyValue(
+      createPaymentBody,
+      "setup_future_usage",
+      "off_session"
+    );
+  });
+
+  afterEach(() => {
+    delete (createPaymentBody as Record<string, unknown>)["setup_future_usage"];
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/02-cards/custom-card-terms.cy.ts]: use removeObjectKey

File: cypress-tests/cypress/e2e/02-cards/default-values-prefill.cy.ts [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+    changeObjectKeyValue(
+      createPaymentBody,
+      "authentication_type",
+      "no_three_ds"
+    );
+    removeObjectKey(createPaymentBody, "billing");
+    changeObjectKeyValue(
+      createPaymentBody,
+      "profile_id",
+      connectorProfileIdMapping.get(connectorEnum.CYBERSOURCE) as string
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/02-cards/default-values-prefill.cy.ts]: `as string` is unnecessary here—please remove it and check other files for similar cases.

File: cypress-tests/cypress/support/component-index.html [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
```
Comment (aritro2002) [inline cypress-tests/cypress/support/component-index.html]: component-index.html and component.js are not used anywhere, either remove it or use it.

File: cypress-tests/cypress/e2e/02-cards/dynamic-fields-comprehensive.cy.ts [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+          country: "US",
+          first_name: "joseph",
+          last_name: "Doe",
+        },
+        phone: {
+          number: "8056594427",
+          country_code: "+91",
+        },
+        email: "test@example.com",
+      });
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/02-cards/dynamic-fields-comprehensive.cy.ts]: You can use `defaultBillingAddress` instead of hardcoding the same billing address.

File: cypress-tests/cypress/support/utils.ts [hunk 13]
```diff
@@ -93,6 +117,25 @@ export const createPaymentBody = {
   },
 };
 
+export const defaultBillingAddress = {
+  email: "hyperswitch_sdk_demo_id@gmail.com",
+  address: {
+    line1: "1467",
+    line2: "Harrison Street",
+    line3: "Harrison Street",
+    city: "San Fransico",
```
Comment (aritro2002) [inline cypress-tests/cypress/support/utils.ts]: correct the spelling,
San Francisco

File: cypress-tests/cypress/e2e/02-cards/dynamic-fields-comprehensive.cy.ts [hunk 14]
```diff
@@ -0,0 +1,317 @@
...
+          state: "California",
+          zip: "94122",
+          country: "US",
+          first_name: "joseph",
+          last_name: "Doe",
+        },
+        phone: {
+          number: "8056594427",
+          country_code: "+91",
+        },
```
Comment (aritro2002) [inline cypress-tests/cypress/e2e/02-cards/dynamic-fields-comprehensive.cy.ts]: use defaultBillingAddress

File: cypress-tests/cypress.config.js [hunk 15]
```diff
@@ -3,8 +3,35 @@ const { defineConfig } = require("cypress");
...
   },
-  retries: { runMode: 1, openMode: 1 },
-});
\ No newline at end of file
+
+  env: {
+    HYPERSWITCH_API_URL: "https://sandbox.hyperswitch.io",
+  },
+
+  // component: {
```
Comment (aritro2002) [inline cypress-tests/cypress.config.js]: remove commented code

File: cypress-tests/cypress/support/types.ts [hunk 16]
```diff
@@ -36,11 +36,16 @@ declare global {
         secretKey: string,
         createPaymentBody: Record<string, any>,
       ): Chainable<Response<any>>;
-      getGlobalState(key: string): Chainable<Response<any>>;
+      getGlobalState(key: string): Chainable<string>;
```
Comment (aritro2002) [inline cypress-tests/cypress/support/types.ts:41]: why?

Comment (preetamrevankar) [reply cypress-tests/cypress/support/types.ts:41]: The `getGlobalState` command retrieves values from a global state object (set by `createPaymentIntent`), not from an HTTP response. The `client_secret` stored is a string, so `Chainable<string>` is the correct return type. The previous `Chainable<Response<any>>` was incorrect.

**Review by aritro2002 — DISMISSED** [review]

File: cypress-tests/cypress/e2e/01-sdk-core/submit-button-states.cy.ts [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+        .find(`[data-testid=${testIds.expiryInputTestId}]`)
+        .safeType(
+          stripeCards.successCard.card_exp_month +
+            stripeCards.successCard.card_exp_year,
+        );
+
+      cy.get("#submit").should("be.visible").click();
+
+      getIframeBody()
+        .find(".Error.pt-1", { timeout: 5000 })
```
Comment (AbhishekChorotiya) [inline cypress-tests/cypress/e2e/01-sdk-core/submit-button-states.cy.ts:166]: it is not checking for CVC specific error, it will return true if any validation error is there in the form

Comment (AbhishekChorotiya) [reply cypress-tests/cypress/e2e/01-sdk-core/submit-button-states.cy.ts:166]: do it for all the cases in this test file

**Review by AbhishekChorotiya — APPROVED** [review]
